src/authutils/user.py

```diff
@@ -11,11 +12,16 @@
 
 
 def set_current_user(**kwargs):
-    default_expected_audience = flask.current_app.config.get("USER_API")
+    logger = get_logger(__name__, log_level="info")
+
+    default_expected_audience = flask.current_app.config.get("OIDC_ISSUER")
```

Comment:
`OIDC_ISSUER` is set in `creds.json` manually from what I can tell. I suppose this is okay but this whole setup is a bit confusing. I believe the aud claim is meant to be a way for a client app to check that a token is meant for them to use, but since this is internal microservices asking if they can use a fence token, it does seem a bit weird that we're even checking this. B/c all the microservices are just saying they're Fence by supply the audience as the `OIDC_ISSUER`

It's like saying "I'm going to check to see if this ID is meant for me to use" but instead of uniquely identifying yourself you ask "is the one who made the ID allowed to use the ID?"

Reply:
> audience
>     OPTIONAL. The logical name of the target service where the client intends to use the requested security token. This serves a purpose similar to the resource parameter but with the client providing a logical name for the target service. Interpretation of the name requires that the value be something that both the client and the authorization server understand. An OAuth client identifier, a SAML entity identifier [[OASIS.saml-core-2.0-os](https://www.rfc-editor.org/rfc/rfc8693#OASIS.saml-core-2.0-os)], and an OpenID Connect Issuer Identifier [[OpenID.Core](https://www.rfc-editor.org/rfc/rfc8693#OpenID.Core)] are examples of things that might be used as audience parameter values. However, audience values used with a given authorization server must be unique within that server to ensure that they are properly interpreted as the intended type of value. Multiple audience parameters may be used to indicate that the issued token is intended to be used at the multiple audiences listed. The audience and resource parameters may be used together to indicate multiple target services with a mix of logical names and resource URIs.

https://www.rfc-editor.org/rfc/rfc8693

Reply:
I think adding "fence-service" to the allowed audiences in tokens would fix this, but if I understand correctly that would allow any Gen3 to use tokens generated by any other Gen3 which is not desirable

Reply:
I think having our services talk to the fence service using `USER_API` configuration that can be set directly to `http://fence-service`, and validate the audience with `OIDC_ISSUER` would be the best balance then. I do see this in some of our services, and I believe that it is how sheepdog/peregrine used to be configured pre-3.9 upgrade.